src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuilding.java

```diff
@@ -0,0 +1,298 @@
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Represents one building order to complete. Has his own structure for the building.
+ */
+public class WorkOrderBuilding extends AbstractWorkOrder
+{
+    private static final String TAG_CUSTOM_NAME            = "customName";
+    private static final String TAG_CUSTOM_PARENT_NAME     = "customParentName";
+    private static final String TAG_PARENT_TRANSLATION_KEY = "parentTranslationKey";
+
+    /**
+     * Maximum distance a builder can have from the building site.
+     */
+    private static final double MAX_DISTANCE_SQ = 100 * 100;
+
+    public static WorkOrderBuilding create(@NotNull final WorkOrderType type, @NotNull final IBuilding building)
+    {
+        int targetLevel = building.getBuildingLevel();
+        switch (type)
+        {
+            case BUILD:
+                targetLevel = 1;
+                break;
+            case UPGRADE:
+                targetLevel++;
+                break;
+            case REMOVE:
+                targetLevel = 0;
+                break;
+        }
+
+        String schematicName;
+        final int targetSchematicLevel = type == WorkOrderType.REMOVE ? building.getBuildingLevel() : targetLevel;
+        final TileEntity buildingTE = building.getColony().getWorld().getBlockEntity(building.getID());
+        if (buildingTE instanceof AbstractTileEntityColonyBuilding)
+        {
+            if (!((AbstractTileEntityColonyBuilding) buildingTE).getSchematicName().isEmpty())
+            {
+                schematicName = ((AbstractTileEntityColonyBuilding) buildingTE).getSchematicName()
+                  .replaceAll("\\d$", "") + targetSchematicLevel;
+            }
+            else
+            {
+                schematicName = building.getSchematicName() + targetSchematicLevel;
+            }
+        }
+        else
+        {
+            schematicName = building.getSchematicName() + targetSchematicLevel;
+        }
+
+        String structureName = new StructureName(Structures.SCHEMATICS_PREFIX, building.getStyle(), schematicName).toString();
+        WorkOrderBuilding wo = new WorkOrderBuilding(
+          structureName,
+          building.getBuildingType().getTranslationKey(),
+          type,
+          building.getID(),
+          building.getRotation(),
+          building.getTileEntity() == null ? building.isMirrored() : building.getTileEntity().isMirrored(),
+          building.getBuildingLevel(),
+          targetLevel);
+        wo.setCustomName(building);
+        return wo;
+    }
+
+    private String customName;
+
+    private String customParentName;

```

Comment:
I'd move those above the static method and javaodc

Reply:
Should I do this consistently across all classes, seeing as I did the same in `AbstractWorkOrder`.
I usually keep all my static members at the top, then class members, unless you rather have a different order.

Reply:
we usually have the fields at the top before the static methods (aside of static { } blocks)